src/runtime/matlab.cpp

```diff
@@ -0,0 +1,377 @@
+#include "HalideRuntime.h"
+
+#define INLINE inline __attribute__((always_inline))
+
+#ifndef MX_API_VER
+#define MX_API_VER 0x07040000
+#endif
+
+struct mxArray;
+
+// It is important to have the mex function pointer definitions in a
+// namespace to avoid silently conflicting symbols with matlab at
+// runtime.
+namespace Halide {
+namespace Runtime {
+namespace mex {
+
+enum { TMW_NAME_LENGTH_MAX = 64 };
+enum { mxMAXNAM = TMW_NAME_LENGTH_MAX };
+
+typedef bool mxLogical;
+typedef int16_t mxChar;
+
+enum mxClassID {
+    mxUNKNOWN_CLASS = 0,
```

Comment:
these enums are an eye-gouging mix of lowercase and uppercase; we should probably go with the convention elsewhere in Halide (all_lowercase_names).